### PR TITLE
Separate download scripts, feedback session

### PR DESCRIPTION
**Pull request by paroma**
* Separate download scripts into full VRD and sample version
* Add content to clarify sample dataset form

### CODE REVIEW

File: scene_graph/download_data.sh [hunk 1]
```diff
@@ -35,23 +35,7 @@ if [ ! -d "scene_graph/data" ]; then
...
-    #     unzip sg_dataset.zip
-    #     rm sg_dataset.zip
-    #     cd ../../..
-    # fi
-
-    # Download and unzip all images
+    # Download and unzip sample images
     mkdir sg_dataset
     cd sg_dataset
     git clone $SAMPLE_IMAGES_URL
```
Comment (vincentschen) [inline scene_graph/download_data.sh]: nit: can we rename this file `download_sample_data.sh` so it's clearer exactly what it's for?

File: scene_graph/download_full_data.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    # Delete the zip files.
+    rm vrd.zip
+    cd VRD
+
+    # Download and unzip all images
+    wget $IMAGES_URL
+    unzip sg_dataset.zip
+    rm sg_dataset.zip
+    cd ../../..
```
Comment (vincentschen) [inline scene_graph/download_full_data.sh]: nit: is there a way to run all these commands from the root directory (e.g. `unzip something/something/sg_dataset.zip`) so we aren't `cd ../../..`? might look cleaner

File: scene_graph/utils.py [hunk 3]
```diff
@@ -55,18 +55,31 @@ def vrd_to_pandas(
...
     NOTE: Only loads semantic relationship examples.
     """
-    try:
-        subprocess.run(
-            ["bash", "scene_graph/download_data.sh"], check=True, stderr=subprocess.PIPE
-        )
-    except subprocess.CalledProcessError as e:
-        print(e.stderr.decode())
-        raise e
+    if sample or is_travis:
```
Comment (vincentschen) [inline scene_graph/utils.py]: nice!

File: scene_graph/vrd_tutorial.py [hunk 4]
```diff
@@ -1,4 +1,19 @@
-# %% [markdown]
+# ---
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: let's remove this metadata chunk

File: scene_graph/vrd_tutorial.py [hunk 5]
```diff
@@ -26,38 +41,45 @@
...
+# +
 # %load_ext autoreload
 # %autoreload 2
 
 import numpy as np
 
-# %%
+# -
+
+# If you are running this notebook for the first time, it will take ~15 mins to download all the required data. **Setting `sample = False` will download the full VRD dataset, which can take ~3 hours!**
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: > required sample data

Comment (vincentschen) [reply scene_graph/vrd_tutorial.py]: can we move the "setting sample = False..." as a comment in the underlying code block, or just move it down? 
people don't know what the `sample` variable is yet

File: scene_graph/vrd_tutorial.py [hunk 6]
```diff
@@ -26,38 +41,45 @@
...
+#
+# The sampled version of the dataset **uses the same 26 examples across the train, dev, and test sets. This setting is meant to demonstrate how Snorkel works with this task, NOT TO DEMONSTRATE PERFORMANCE.**
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: nit: can we use less bold? bold + caps seems like a lot :)
maybe just italicize the caps part?

File: scene_graph/vrd_tutorial.py [hunk 7]
```diff
@@ -26,38 +41,45 @@
...
+#
+
+# +
 from scene_graph.utils import load_vrd_data
 
-train_df, valid_df, test_df = load_vrd_data()
+sample = True
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: see comments above about commenting this part

File: scene_graph/vrd_tutorial.py [hunk 8]
```diff
@@ -26,38 +41,45 @@
...
+is_travis = "TRAVIS" in os.environ
+train_df, valid_df, test_df = load_vrd_data(sample, is_travis)
 
 print("Train Relationships: ", len(train_df))
 print("Dev Relationships: ", len(valid_df))
 print("Test Relationships: ", len(test_df))
+# -
+
+# Note that the training DataFrame will have a labels field with all -1s. This denotes the lack of labels for that particular dataset. In this tutorial, we will assign probabilistic labels to the training set by writing labeling functions over attributes of the subject and objects!
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: nit: format backtick DataFrame backtick

File: scene_graph/vrd_tutorial.py [hunk 9]
```diff
@@ -26,38 +41,45 @@
...
 
-# %% [markdown]
 # ## 2. Writing Labeling Functions
 # We now write labeling functions to detect what relationship exists between pairs of bounding boxes. To do so, we can encode various intuitions into the labeling functions. _Categorical_ intution: knowledge about the categories of subjects and objects usually involved in these relationships (e.g., `person` is usually the subject for predicates like `ride` and `carry`), and _spatial_ intuition: knowledge about the relative positions of the subject and objects (e.g., subject is usually higher than the object for the predicate `ride`).
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: nit: can we break up _categorical intuition_ and _spatial intuition_ into bullets?

File: scene_graph/vrd_tutorial.py [hunk 10]
```diff
@@ -155,36 +179,33 @@ def LF_area(x):
...
-# %%
+# +
 from snorkel.labeling import LabelModel
 
 label_model = LabelModel(cardinality=3, verbose=True)
 label_model.fit(L_train, seed=123, lr=0.01, log_freq=10, n_epochs=100)
+# -
 
-# %% [markdown]
 # We use [F1](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html) Micro average for the multiclass setting, which calculates metrics globally by counting the total true positives, false negatives and false positives.
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: nit: which calculate metrics globally, "across classes", ...

File: scene_graph/vrd_tutorial.py [hunk 11]
```diff
@@ -155,36 +179,33 @@ def LF_area(x):
...
 # ## 4. Train a Classifier
 # You can then use these training labels to train any standard discriminative model, such as [an off-the-shelf ResNet](https://github.com/KaimingHe/deep-residual-networks), which should learn to generalize beyond the LF's we've developed!
 
-# %% [markdown]
 # #### Create DataLoaders for Classifier
 
-# %%
+# +
 from snorkel.classification import DictDataLoader
 from scene_graph.model import FlatConcat, SceneGraphDataset, WordEmb, init_fc
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: can we handle the below `TRAIN_DIR` assignment in an `if sample` now?

File: scene_graph/vrd_tutorial.py [hunk 12]
```diff
@@ -269,6 +290,6 @@ def LF_area(x):
     checkpointer_config={"checkpoint_dir": "checkpoint"},
 )
 trainer.fit(model, [train_dl])
+# -
 
-# %%
 model.score([valid_dl])
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: let's add a final few sentences about what we accomplished in the tutorial!

File: scene_graph/download_full_data.sh (hunk 2 above)
Comment (paroma) [reply scene_graph/download_full_data.sh]: I tried that originally and I couldn't get the unzip results to land up in the correct location somehow, will make a note for later though

File: scene_graph/vrd_tutorial.py (hunk 11 above)
Comment (paroma) [reply scene_graph/vrd_tutorial.py]: good catch!

File: scene_graph/vrd_tutorial.py [hunk 13]
```diff
@@ -155,43 +181,43 @@ def LF_area(x):
...
 
-# change to "scene_graph/data/VRD/sg_dataset/sg_train_images" for full set
-TRAIN_DIR = "scene_graph/data/VRD/sg_dataset/samples"
 train_df["labels"] = label_model.predict(L_train)
 
+if sample:
+    TRAIN_DIR = "scene_graph/data/VRD/sg_dataset/samples"
+else:
+    TRAIN_DIR = "scene_graph/data/VRD/sg_dataset/sg_train_images"
+
```
Comment (vincentschen) [inline scene_graph/vrd_tutorial.py]: nit: could be cleaner:

`TRAIN_DIR = "scene_graph/data/VRD/sg_dataset/samples" if sample else "scene_graph/data/VRD/sg_dataset/sg_train_images"`

Comment (paroma) [reply scene_graph/vrd_tutorial.py]: the line becomes too long and then looks odd sadly

**Comment by paroma** [comment]
stacked with #66

**Pull request merged**